Build failure: limine

### Nixpkgs version

- Unstable (25.05)

### Steps to reproduce

- Simply try to build it (nix build nixpkgs#limine)
- Failure

### Can Hydra reproduce this build failure?

Hydra is currently rebuilding this package.

### Link to Hydra build job

_No response_

### Relevant log output

```console
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/wkavw35bpff486sxvdkslsca6cka9wkw-limine-8.7.0.tar.gz
source root is limine-8.7.0
setting SOURCE_DATE_EPOCH to timestamp 1736517482 of file "limine-8.7.0/version"
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Updating Autotools / GNU config script to a newer upstream version: ./build-aux/config.sub
Updating Autotools / GNU config script to a newer upstream version: ./build-aux/config.guess
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
patching script interpreter paths in ./configure
./configure: interpreter directive changed from "#! /bin/sh" to "/nix/store/4fvc5fm8bszmkydng1ivrvr5cbvr1g60-bash-5.2p37/bin/sh"
configure flags: --prefix=/nix/store/0y26p1wzadm6c5j0xg57m8nrmbansg3l-limine-8.7.0 --bindir=/nix/store/0y26p1wzadm6c5j0xg57m8nrmbansg3l-limine-8.7.0/bin --sbindir=/nix/store/0y26p1wzadm6c5j0x>
checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-pc-linux-gnu
checking for gcc... clang
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether the compiler supports GNU C... yes
checking whether clang accepts -g... yes
checking for clang option to enable C11 features... none needed
checking for a race-free mkdir -p... /nix/store/pw26z7msfmv291jq1qpp8bbaprs5hkfk-coreutils-9.5/bin/mkdir -p
checking for a BSD-compatible install... /nix/store/pw26z7msfmv291jq1qpp8bbaprs5hkfk-coreutils-9.5/bin/install -c
checking for a sed that does not truncate output... /nix/store/v93bdqf52x2r2lqijv0f7x3yhylwy957-gnused-4.9/bin/sed
checking for grep that handles long lines and -e... /nix/store/01dri6m7g4bx0kkd39pkgfpzpj3i7y86-gnugrep-3.11/bin/grep
checking for gawk... gawk
checking for find... yes
checking for strip... yes
checking for stdio.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for strings.h... yes
checking for sys/stat.h... yes
checking for sys/types.h... yes
checking for unistd.h... yes
checking for stdint.h... (cached) yes
checking for stddef.h... yes
checking for stdbool.h... yes
checking for limits.h... yes
checking for inttypes.h... (cached) yes
checking for stdio.h... (cached) yes
checking for stdlib.h... (cached) yes
checking for string.h... (cached) yes
...skipping...
checking for find... yes
checking for strip... yes
checking for stdio.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for strings.h... yes
checking for sys/stat.h... yes
checking for sys/types.h... yes
checking for unistd.h... yes
checking for stdint.h... (cached) yes
checking for stddef.h... yes
checking for stdbool.h... yes
checking for limits.h... yes
checking for inttypes.h... (cached) yes
checking for stdio.h... (cached) yes
checking for stdlib.h... (cached) yes
checking for string.h... (cached) yes
checking for llvm-ar (x86_64)... yes
checking for clang (x86_64)... yes
checking for ld.lld (x86_64)... yes
checking for llvm-objcopy (x86_64)... yes
checking for llvm-objdump (x86_64)... yes
checking for nasm... yes
configure: creating ./config.status
config.status: creating man/man1/limine.1
config.status: creating GNUmakefile
config.status: creating config.h
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
build flags: -j16 SHELL=/nix/store/4fvc5fm8bszmkydng1ivrvr5cbvr1g60-bash-5.2p37/bin/bash
mkdir -p '/build/limine-8.7.0/bin'
cp '/build/limine-8.7.0/host/host.mk' '/build/limine-8.7.0/bin/Makefile'
cp '/build/limine-8.7.0/host/.gitignore' '/build/limine-8.7.0/bin/'
make all1
make[1]: Entering directory '/build/limine-8.7.0'
make common-uefi-x86-64
make[2]: Entering directory '/build/limine-8.7.0'
TOOLCHAIN_FILE='/build/limine-8.7.0/toolchain-files/uefi-x86_64-toolchain.mk' \
TARGET=uefi-x86-64 \
BUILDDIR='/build/limine-8.7.0/common-uefi-x86-64' \
        make -C '/build/limine-8.7.0/common' -f common.mk
make[3]: Entering directory '/build/limine-8.7.0/common'
make[3]: *** No rule to make target 'supplying', needed by 'Warning'.  Stop.
make[3]: Leaving directory '/build/limine-8.7.0/common'
make[2]: *** [GNUmakefile:343: common-uefi-x86-64] Error 2
make[2]: Leaving directory '/build/limine-8.7.0'
make[1]: *** [GNUmakefile:239: limine-uefi-x86-64] Error 2
make[1]: Leaving directory '/build/limine-8.7.0'
make: *** [GNUmakefile:96: all] Error 2
lines 33-83/83 (END)
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.65-rt47, NixOS, 25.05 (Warbler), 25.05.20250126.852ff1d`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.11`
 - channels(root): `"nixos-23.05"`
 - nixpkgs: `/nix/store/wshnc0kqk1qz7iffb1yqri8a5cy6v7w5-source`

### Notify maintainers

@48cf @phip1611 @surfaceflinger 
---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/li/limine/package.nix
+++ b/pkgs/by-name/li/limine/package.nix
@@ -47,7 +47,13 @@ let
     ++ lib.optionals pxeSupport [ "--enable-bios-pxe" ]
     ++ lib.concatMap uefiFlags (
       if targets == [ ] then [ stdenv.hostPlatform.parsed.cpu.name ] else targets
-    );
+    )
+    ++ [
+      "TOOLCHAIN_FOR_TARGET=llvm"
+      # `clang` on `PATH` has to be unwrapped, but *a* wrapped clang
+      # still needs to be available
+      "CC=${lib.getExe stdenv.cc}"
+    ];
 in
 
 assert lib.assertMsg (!(biosSupport && !hasI686)) "BIOS builds are possible only for x86";
@@ -76,6 +82,7 @@ stdenv.mkDerivation {
 
   nativeBuildInputs =
     [
+      llvmPackages.clang-unwrapped
       llvmPackages.libllvm
       llvmPackages.lld
     ]

